src/main/java/OrderRepo.java

```diff
@@ -0,0 +1,27 @@
+import javax.swing.text.html.Option;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+
+public class OrderRepo {
+    private HashMap<String, Order> orders = new HashMap<>();
+
+    public OrderRepo(List<Order> orders){
+        for(Order o : orders) this.orders.putIfAbsent(o.getId(), o);
+    }
+
+    public Optional<Order> get(String id){
+        Optional<Order> orderReturn = Optional.of(orders.get(id));
```

Comment:
Falls `orders.get(id)` `null` zurückgibt, wirft `Optional.of(orders.get(id))` eine `NullPointerException`. Darum würde ich `Optional.ofNullable` verwenden.